hknweb/academics/models/base_models.py

```diff
@@ -1,5 +1,30 @@
 from django.db import models
+from django.db.models import Count
+
 
 class AcademicEntity(models.Model):
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+    """
+    Merges two Academic entities of the same type.
+    to_keep: the entity which should subsume the other one (take on all its children), and whose parent should remain
+    to_remove: the entity which should be removed 
+    """
+    @staticmethod
+    def merge(to_keep, to_remove):
+        if len(to_remove) == 0:
+            raise Exception()
+        child_classes=to_remove[0].CHILDREN
+        for child_class in child_classes:
+            affected_children = getattr(to_remove[0],child_class+"_set")
+            for dup in to_remove[1:]:

```

Comment:
Consider using the `reduce` function e.g. https://www.geeksforgeeks.org/python-bitwise-or-among-list-elements/

Reply:
Hey Brian, looked at that link, apparently reduce was removed in python3 because its not as readable as just using a for loop.

Reply:
Try [`functools.reduce`](https://docs.python.org/3/library/functools.html#functools.reduce)? Just thought it'd might be cleaner and easier to understand (funnily enough).